runtime/src/deterministic.rs

```diff
@@ -747,15 +757,38 @@ impl crate::Spawner for Context {
 
         // Set up the task
         let executor = self.executor.clone();
+
         move |f: F| {
-            let (f, handle) = Handle::init_future(f, gauge, false);
+            // Give spawned task its own empty children list
+            let (f, handle) =
+                Handle::init_future(f, gauge, false, Arc::new(Mutex::new(Vec::new())));
 
             // Spawn the task
             Tasks::register_work(&executor.tasks, label, Box::pin(f));
             handle
         }
     }
 
+    fn spawn_child<F, Fut, T>(&self, f: F) -> Handle<T>
+    where
+        F: FnOnce(Self) -> Fut + Send + 'static,
+        Fut: Future<Output = T> + Send + 'static,
+        T: Send + 'static,
+    {
+        // Create child context
+        let child_context = self.clone();
```

Comment:
I actually prefer requiring `spawn_child` consume `self` here (like a normal context spawn).

It should be up to the user to ensure they clone the context for any child (this also allows them to provide a `with_label` here.